### PR TITLE
Accept SDK version above 3.0.

### PR DESCRIPTION
**Pull request by lrhn**
Fix local variable typos and add comments preparing for 3.0 migration.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
+## 1.17.2
+
+* Accept Dart SDK versions above 3.0.
```
Comment (kevmoo) [inline CHANGELOG.md:3]: but also drop SDK req to 2.17?

File: lib/src/boollist.dart [hunk 2]
```diff
@@ -42,18 +42,18 @@ abstract class BoolList with ListMixin<bool> {
   factory BoolList(int length, {bool fill = false, bool growable = false}) {
     RangeError.checkNotNegative(length, 'length');
 
-    BoolList boolist;
+    BoolList boolList;
```
Comment (kevmoo) [inline lib/src/boollist.dart:45]: careful about unrelated changes

**Review by kevmoo — CHANGES_REQUESTED** [review]
Drop the CI run to run on 2.17 – or put but the SDK constraint

File: lib/src/boollist.dart (hunk 2 above)
Comment (lrhn) [reply lib/src/boollist.dart:45]: I'm very careful ... to not forget to fix them.

File: CHANGELOG.md (hunk 1 above)
Comment (lrhn) [reply CHANGELOG.md:3]: That was a mistake. Don't know how. Fixed.

**Review by kevmoo — APPROVED** [review]
LGTM
